tests/mapreduce/test.mapreduce.js

```diff
@@ -3697,5 +3697,15 @@ function tests(suiteName, dbName, dbType, viewType) {
         });
       });
     });
+
+    it('should throw when user supplies map func unallowed by strict mode (#6234)', function () {
+      var db = new PouchDB(dbName);
+      return db.query("(function () { globalVar = 'hello'; })()")
+      .then(function (res) {
+        should.not.exist(res);
+      }).catch(function (err) {
+        err.message.should.be.a('string');
+      });
```

Comment:
Technically this would need to be a second-arg error handler (i.e. `then(success, failure)`) rather than a `.catch()`, because if `should.not.exist()` throws then you end up in the `.catch()` function, which is not what we want. There are some details in [this blog post](https://pouchdb.com/2015/05/18/we-have-a-problem-with-promises.html); it's subtle.